components/profile/my-network-settings/tokens-settings.tsx

```diff
@@ -58,11 +58,11 @@ export default function TokensSettings({
         dbTransactionalAllowed
       } = dbTokens.reduce((previous, current) => {
         const tmp = { ...previous };
-        const { isTransactional, isAllowed } = current;
+        const { isTransactional, isAllowed, isReward } = current;
```

Comment:
this code can be simplified now that we have the `isReward`, something like this:

```ts
const dbTokens = await getTokens();
const availableRewardList = [];
const availableTransactionList = [];

for (const token of dbTokens) {
  if (token.isReward)
    availableRewardList.push(token);
  else if (token.isTransactional)
    availableTransactionList.push(token);
}

if (isGovernorRegistry)  {
  /** unsure if this is needed */
  setSelectedRewardTokens(availableRewardList);
  setSelectedTransactionalTokens(availableTransactionList);
  /** but kept it because previous code had it. */

  availableRewardList.push(...availableTransactionList);
  availableTransactionList.push(...availableRewardList)
}

setAllowedRewardTokensList(availableRewardList)
setAllowedTransactionalTokensList(availableTransactionList)


```

Reply:
I tried to apply this logic, but there were some use cases it didn't cover. how we use this component for governance and registry. I chose to keep and adjust only what was necessary